app/helpers/pieces_helper.rb

```diff
@@ -0,0 +1,14 @@
+module PiecesHelper
+  def castling(destination_piece)
```

Comment:
I think it definitely made sense before. If you do the refactor in my other comment, then it will be really simple & probably not worth extracting.

Could look like.

```
if destination_piece.present? && current_piece.can_castle?(destination_piece)
  result = current_piece.move_to!(params[:piece][:x], params[:piece][:y])
end
```